### PR TITLE
Finish up typing meshmode.mesh.processing

### PR DESCRIPTION
**Pull request by alexfikl**
Following #357.

There are some extra changes in here for some functions that weren't handling the `group.vertex_indices == None` case, which `mypy` started complaining about once the functions were typed.

### CODE REVIEW

File: meshmode/mesh/processing.py [hunk 1]
```diff
@@ -961,7 +950,7 @@ def split_mesh_groups(
             is_conforming=mesh.is_conforming)
 
     if return_subgroup_mapping:
-        return mesh, subgroup_to_group_map
+        return mesh, subgroup_to_group_map      # type: ignore[return-value]
```
Comment (inducer) [inline meshmode/mesh/processing.py:953]: We could instead be honest and declare the return type to be a `Union`. Would that work?

File: meshmode/mesh/processing.py [hunk 2]
```diff
@@ -1314,11 +1316,11 @@ def map_mesh(mesh: Mesh, f: Callable[[np.ndarray], np.ndarray]) -> Mesh:
...
     if mesh._facial_adjacency_groups is not None:
         has_adj_maps = any([
-            hasattr(fagrp, "aff_map")
-            and (fagrp.aff_map.matrix is not None
-                or fagrp.aff_map.offset is not None)
+            fagrp.aff_map.matrix is not None or fagrp.aff_map.offset is not None
             for fagrp_list in mesh.facial_adjacency_groups
-            for fagrp in fagrp_list])
+            for fagrp in fagrp_list
+            if isinstance(fagrp, (InteriorAdjacencyGroup, InterPartAdjacencyGroup))
```
Comment (inducer) [inline meshmode/mesh/processing.py:1322]: This should include a check to ensure that not-known classes aren't sliently not considered.

**Review by inducer — COMMENTED** [review]
Thanks! Two wrinkles that are, hopefully, easy to address.

File: meshmode/mesh/processing.py (hunk 1 above)
Comment (alexfikl) [reply meshmode/mesh/processing.py]: Fixed in https://github.com/inducer/meshmode/compare/47349894ac4cd5d44460499a3b1e61231b4d86ee..8b5b8ac6c578291aa5cb55a5805ca740ae8d6a65

File: meshmode/mesh/processing.py (hunk 2 above)
Comment (alexfikl) [reply meshmode/mesh/processing.py]: I reverted back to the original version that only checked `hasattr` 
https://github.com/inducer/meshmode/compare/47349894ac4cd5d44460499a3b1e61231b4d86ee..8b5b8ac6c578291aa5cb55a5805ca740ae8d6a65

Would you prefer to change that and use a stricter check?

Comment (inducer) [reply meshmode/mesh/processing.py]: I think `hasattr` is OK.

**Comment by inducer** [comment]
Thanks!

**Pull request merged**